File: Dimension/Model/OutgoingConnection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Dimension.Model
{
    public abstract class OutgoingConnection : Connection
    {
        public ulong rate;
        public delegate void CommandReceived(Commands.Command c);
        public abstract event CommandReceived commandReceived;
        public abstract void send(Commands.Command c);
        public abstract bool connected
        {
            get;
        }
    }
}

File: lib/model/outgoing_connection.dart
import 'dart:async';
import 'connection.dart';
import 'commands/command.dart';

abstract class OutgoingConnection extends Connection {
  bool rateLimiterDisabled = false;
  int rate = 0;
  void Function(Command c)? commandReceived;
  Future<void> send(Command c);
  bool get connected;
}
